### DISCUSSION TITLE
How about add a `>>>` operator which alias `via`?

### DISCUSSION THREAD

**Discussion by He-Pin** [apache/pekko#969] [Ideas]
ZIO is using `>>>` alias with `via`  and I think it's very readable .
And FS2 is using `>>` alias with `flatmap(s: => Stream)` too.

There is already a `++` in scaladsl.

And there is a `  def >>>(e: Enumerator[E]): Enumerator[E] = andThen(e)` in https://github.com/playframework/play-iteratees

**Comment by pjfanning** [comment]
I'm not a fan of function names like this. I'll go with the consensus but my preference is to stick with readable names. Even if not everyone is fluent in English, you can use a dictionary. No dictionary will tell you what `>>>` means.

**Reply by laglangyue** [reply]
![image](https://github.com/apache/incubator-pekko/assets/35491928/c4cf869a-9254-4286-90bf-9c99de770c11)
Is it necessary to unify symbols? Using `~>` is pekko's style.
When we customize the Graph, we will learn this symbol `~>`. If we have already learned this symbol by using it  with source/flow, it will be very helpful for us to learning the Graph.And we don't have to define too many symbols

**Reply by mdedetrich** [reply]
> I'm not a fan of function names like this. I'll go with the consensus but my preference is to stick with readable names. Even if not everyone is fluent in English, you can use a dictionary. No dictionary will tell you what `>>>` means.

This is probably not a well known feature, but scala doc automatically come with a hoogle like search feature i.e. if you go to https://pekko.apache.org/api/pekko/current/ you will find a search bar at top and you can search method names such as [flatMapPrefix](https://pekko.apache.org/api/pekko/current/?search=flatMapPrefix) or [via](https://pekko.apache.org/api/pekko/current/?search=via) however embarrassingly symbols like [`~>`](https://pekko.apache.org/api/pekko/current/?search=~%3E) is not working likely due to how the stream API is structured (we should probably file an issue for this).

Personally I am not too fussed on this as `>>>` is quite common in idiomatic Scala now and the `scaladsl` should reflect that.  With `javadsl` its obviously a different story.

**Reply by He-Pin** [reply]
There is a `>>` in fs2. and @laglangyue I think we can not use this `~>/<~` , as these methods  defined in `GraphDSL.implicits` too.

**Reply by mdedetrich** [reply]
`>>>` and `~>` fundamentally mean different things as well. `~>` is about joinings nodes in a graph where as `>>>` is an alias for `flatMap`.

**Reply by He-Pin** [reply]
noop, `>>` is `flatmap`, and `>>>` is `via`.

I'm asking for this because I find it will looks nicer when using it with `for`.
eg:
```scala
import GraphDSL.implicits._
for {
a <- (flowA >>> flowB >>> flowC)
b <- flowD >>> Source(...)
} yield b
```

**Reply by mdedetrich** [reply]
> noop, `>>` is `flatmap`, and `>>>` is `via`.

Yeah I had a brain fart, I meant `via`. But now that you mention it, maybe it also makes sense to add `>>` if you don't want to use for comprehension?

**Reply by laglangyue** [reply]
fine, I also enjoy using symbols in my code.
It seems that there are not many people discussing on github. I hope you can investigate it in the mailing list. I will `+1` for you

**Reply by laglangyue** [reply]
Many operators in fs2 have the symbols. Do you plan to implement all of them, or just this two?
If many symbols, it is best to discuss it together to avoid future ambiguity and sysbols  abuse.

**Reply by jxnu-liguobin** [reply]
My point of view is that although I sometimes like to use it, I choose to oppose it. Peeko is not an FP library. This results in beginners needing to look at the documentation to understand.

**Reply by He-Pin** [reply]
@jxnu-liguobin But you are using ZIO with `>>>` at $Work ASAIK :)

Pekko Stream is functional and it will not run before we call the `run`, and the Effect is bind to `Future` not `F[_]` in fs2 and `ZIO` in ZIO.

**Reply by jxnu-liguobin** [reply]
> @jxnu-liguobin But you are using ZIO with `>>>` at $Work ASAIK :)
> 
> 
> 
> Pekko Stream is functional and it will not run before we call the `run`, and the Effect is bind to `Future` not `F[_]` in fs2 and `ZIO` in ZIO.

The most commonly used operator in zio is the flatmap zip operator
.  These are already difficult to remember to a certain extent.
`*>  <*  <+> <&>`
Sometimes I even forget.

**Reply by naosense** [reply]
Agree, too many symbols are difficult to use, just use symbols for conventional operations
